src/org/parosproxy/paros/core/proxy/ProxyThread.java

```diff
@@ -451,13 +452,19 @@ protected void processHttp(HttpRequestHeader requestHeader, boolean isSecure) th
 		        	return;
 			    }
 			}
-
-			if (parentServer.isEnableApi() &&
-					API.getInstance().handleApiRequest(requestHeader, httpIn, httpOut, isRecursive(requestHeader))) {
-				// It was an API request
-				return;
+			if (parentServer.isEnableApi()) {
+				msg = API.getInstance().handleApiRequest(requestHeader, httpIn, httpOut, isRecursive(requestHeader));
+				if (msg != null) {

```

Comment:
Should have another if (to just return) for when the `msg` is empty? We shouldn't notify if the API request was not allowed or is already handled.